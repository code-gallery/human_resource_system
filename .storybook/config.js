import { configure } from '@storybook/react'

const loadStories = () => {
  require('../src/stories')
}

configure(loadStories, module)
