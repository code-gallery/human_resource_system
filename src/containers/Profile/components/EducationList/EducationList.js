import React from 'react'
import PropTypes from 'prop-types'
import EducationItem from './components/EducationItem'
import EducationItemEdit from './components/EducationItemEdit'

class EducationList extends React.Component {
  constructor(props) {
    super(props)
    this.state = {
      blockIsEditing: false,
      addNew: false,
      reorder: false
    }
  }

  toggleBlockEditing = () => {
    this.setState({
      blockIsEditing: !this.state.blockIsEditing
    })
  }

  toggleAddNewItem = () => {
    this.setState({
      addNew: !this.state.addNew
    })
  }

  render() {
    const { education, editMode, reorder, awards,
      qualifications, toggleReorder, deleteEducation,
      saveEntity, filterEntries, highSchoolDegrees, user
    } = this.props
    const { addNew } = this.state

    return (
      <section className="profile-experience-wrapper">
        {
          editMode &&
          <div className="profile-experience-block-btns">
            <button
              className="reorder-btn"
              onClick={toggleReorder}>
              &nbsp;
            </button>
            <button
              className="add-btn"
              onClick={this.toggleAddNewItem}
            >
              &nbsp;
            </button>
          </div>
        }
        <p className="profile-block-title">Education</p>
        {!reorder &&
          <div>
            {
              editMode && addNew &&
              <EducationItemEdit
                toggleEdit={this.toggleAddNewItem}
                awards={awards}
                qualifications={qualifications}
                highSchoolDegrees={highSchoolDegrees}
                deleteEducation={deleteEducation}
                saveEntity={saveEntity}
                user={user}
              />
            }
            {
              education
                .filter(x => filterEntries(x))
                .map((item, idx) => (
                  <EducationItem
                    {...item}
                    editMode={editMode}
                    toggleBlockEditing={this.toggleBlockEditing}
                    key={idx}
                    awards={awards}
                    qualifications={qualifications}
                    highSchoolDegrees={highSchoolDegrees}
                    deleteEducation={deleteEducation}
                    saveEntity={saveEntity}
                    user={user}
                  />
                ))
            }
          </div>
        }
      </section>
    )
  }
}

EducationList.propTypes = {
  education: PropTypes.array,
  editMode: PropTypes.bool,
  reorder: PropTypes.bool,
  awards: PropTypes.array,
  qualifications: PropTypes.array,
  toggleReorder: PropTypes.func.isRequired,
  deleteEducation: PropTypes.func.isRequired,
  saveEntity: PropTypes.func.isRequired,
  filterEntries: PropTypes.func,
  user: PropTypes.object,
  highSchoolDegrees: PropTypes.array
}

export default EducationList
