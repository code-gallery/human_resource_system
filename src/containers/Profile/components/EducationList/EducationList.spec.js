import React from 'react'
import renderer from 'react-test-renderer'
import { shallow } from 'enzyme'
import { Provider } from 'react-redux'
import { BrowserRouter as Router } from 'react-router-dom'
import config from 'store'
import EducationList from './index'

describe('containers/Profile/components/EducationList', () => {
  const commonProps = {
    reorder: false,
    toggleReorder: jest.fn(),
    deleteEducation: jest.fn(),
    saveEntity: jest.fn(),
    filterEntries: function() {
      return true
    },
    awards: [
      { id: 1, name: 'Qualification (Award)', value1: 'AS-Levels', value2: 'A*' },
      { id: 1, name: 'Qualification (Award)', value1: 'GCSE', value2: 'A*' },
      { id: 2, name: 'Qualification (Award)', value1: 'Master', value2: 'Pass' },
      { id: 3, name: 'Qualification (Award)', value1: 'Doctorate', value2: 'Pass' }
    ],
    highSchoolDegrees: [
      { id: 10, name: 'Qualification (Type)', value1: 'Secondary', value2: 'AS-Levels' },
      { id: 11, name: 'Qualification (Type)', value1: 'Secondary', value2: 'A-Levels' },
      { id: 12, name: 'Qualification (Type)', value1: 'Secondary', value2: 'GCSE' }
    ],
    qualifications: [
      { id: 20, name: 'Qualification (Type)', value1: 'Masters', value2: 'LLM' },
      { id: 21, name: 'Qualification (Type)', value1: 'Masters', value2: 'MSc' },
      { id: 22, name: 'Qualification (Type)', value1: 'Masters', value2: 'MA' }
    ],
    user: {
      mobile_address: true,
      biometrics_status: 'complete'
    },
    education: [
      {
        id: 127,
        verified_status: 'verified',
        institution: 'Birmingham City University',
        org: {
          id: 140,
          name: 'Birmingham City University'
        },
        organisation_id: 140,
        start_date: '2013-09-01T00:00:00.000Z',
        end_date: '2014-07-01T00:00:00.000Z',
        type: 'secondary'
      },
      {
        id: 128,
        verified_status: 'pending_verification',
        institution: 'Oxford Brookes University',
        org: {
          id: 953,
          name: 'Oxford Brookes University'
        },
        organisation_id: 953,
        start_date: '2011-09-01T00:00:00.000Z',
        end_date: '2012-07-01T00:00:00.000Z',
        type: 'secondary'
      }
    ]
  }
  const { store } = config

  describe('when props.editMode is false', () => {
    describe('when sorting by end_date', () => {
      it('renders correctly', () => {
        const props = {
          ...commonProps,
          editMode: false
        }
        const tree = renderer.create(
          <Provider store={store}>
            <Router>
              <EducationList {...props} />
            </Router>
          </Provider>
        ).toJSON()

        expect(tree).toMatchSnapshot()
      })
    })
  })

  describe('when props.editMode is true', () => {
    describe('when sorting by end_date', () => {
      it('renders correctly', () => {
        const props = {
          ...commonProps,
          editMode: true
        }
        const tree = renderer.create(
          <Provider store={store}>
            <Router>
              <EducationList {...props} />
            </Router>
          </Provider>
        ).toJSON()

        expect(tree).toMatchSnapshot()
      })
    })
  })

  describe('when props.editMode = true and props.addNew = true', () => {
    describe('when sorting by end_date', () => {
      it('renders correctly', () => {
        const props = {
          ...commonProps,
          editMode: true,
          addNew: true
        }
        const tree = renderer.create(
          <Provider store={store}>
            <Router>
              <EducationList {...props} />
            </Router>
          </Provider>
        ).toJSON()

        expect(tree).toMatchSnapshot()
      })
    })
  })

  describe('<EducationList />', () => {
    let wrapper
    beforeEach(() => {
      const props = {
        ...commonProps,
        editMode: false
      }

      wrapper = shallow(
        <EducationList {...props} />
      )
    })

    describe('toggleBlockEditing', () => {
      it('updates state correctly', () => {
        expect(wrapper.state('blockIsEditing')).toEqual(false)
        wrapper.instance().toggleBlockEditing()
        expect(wrapper.state('blockIsEditing')).toEqual(true)
      })
    })

    describe('toggleAddNewItem', () => {
      it('updates state correctly', () => {
        expect(wrapper.state('addNew')).toEqual(false)
        wrapper.instance().toggleAddNewItem()
        expect(wrapper.state('addNew')).toEqual(true)
      })
    })
  })
})
