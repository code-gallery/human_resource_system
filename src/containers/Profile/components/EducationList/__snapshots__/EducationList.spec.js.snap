// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`containers/Profile/components/EducationList when props.editMode = true and props.addNew = true when sorting by end_date renders correctly 1`] = `
<section
  className="profile-experience-wrapper"
>
  <div
    className="profile-experience-block-btns"
  >
    <button
      className="reorder-btn"
      onClick={[Function]}
    >
       
    </button>
    <button
      className="add-btn"
      onClick={[Function]}
    >
       
    </button>
  </div>
  <p
    className="profile-block-title"
  >
    Education
  </p>
  <div>
    <div>
      <div
        className="profile-experience-block profile-verified-status"
      >
        <div
          className="profile-experience-btns"
        >
          <button
            className="edit-btn"
            onClick={[Function]}
          >
             
          </button>
        </div>
        <div
          className="relative"
        >
          <div
            className="Avatar Avatar-white Avatar-size-109"
          >
            <img
              alt="logo"
              src="appii-placeholder.png"
            />
          </div>
          <p
            className="visible-xs visible-sm"
          >
            <i
              className="profile-status-ico tooltip-position-top tooltip-movable tooltip-success"
              data-tooltip="Verified"
            >
               
            </i>
          </p>
        </div>
        <div
          className="profile-experience-info"
        >
          <i
            className="profile-status-ico tooltip-position-top tooltip-movable tooltip-success hidden-xs hidden-sm"
            data-tooltip="Verified"
          >
             
          </i>
          <p>
            <a
              href="/organisations/140"
              onClick={[Function]}
              title="Birmingham City University"
            >
              Birmingham City University
            </a>
          </p>
          <span>
            
          </span>
          <em>
            2013
             - 
            2014
          </em>
        </div>
      </div>
    </div>
    <div>
      <div
        className="profile-experience-block profile-pending-status"
      >
        <div
          className="profile-experience-btns"
        >
          <button
            className="edit-btn"
            onClick={[Function]}
          >
             
          </button>
        </div>
        <div
          className="relative"
        >
          <div
            className="Avatar Avatar-white Avatar-size-109"
          >
            <img
              alt="logo"
              src="appii-placeholder.png"
            />
          </div>
          <p
            className="visible-xs visible-sm"
          >
            <i
              className="profile-status-ico tooltip-position-top tooltip-movable tooltip-warning"
              data-tooltip="Pending verification"
            >
               
            </i>
          </p>
        </div>
        <div
          className="profile-experience-info"
        >
          <i
            className="profile-status-ico tooltip-position-top tooltip-movable tooltip-warning hidden-xs hidden-sm"
            data-tooltip="Pending verification"
          >
             
          </i>
          <p>
            <a
              href="/organisations/953"
              onClick={[Function]}
              title="Oxford Brookes University"
            >
              Oxford Brookes University
            </a>
          </p>
          <span>
            
          </span>
          <em>
            2011
             - 
            2012
          </em>
        </div>
      </div>
    </div>
  </div>
</section>
`;

exports[`containers/Profile/components/EducationList when props.editMode is false when sorting by end_date renders correctly 1`] = `
<section
  className="profile-experience-wrapper"
>
  <p
    className="profile-block-title"
  >
    Education
  </p>
  <div>
    <div>
      <div
        className="profile-experience-block profile-verified-status"
      >
        <div
          className="relative"
        >
          <div
            className="Avatar Avatar-white Avatar-size-109"
          >
            <img
              alt="logo"
              src="appii-placeholder.png"
            />
          </div>
          <p
            className="visible-xs visible-sm"
          >
            <i
              className="profile-status-ico tooltip-position-top tooltip-movable tooltip-success"
              data-tooltip="Verified"
            >
               
            </i>
          </p>
        </div>
        <div
          className="profile-experience-info"
        >
          <i
            className="profile-status-ico tooltip-position-top tooltip-movable tooltip-success hidden-xs hidden-sm"
            data-tooltip="Verified"
          >
             
          </i>
          <p>
            <a
              href="/organisations/140"
              onClick={[Function]}
              title="Birmingham City University"
            >
              Birmingham City University
            </a>
          </p>
          <span>
            
          </span>
          <em>
            2013
             - 
            2014
          </em>
        </div>
      </div>
    </div>
    <div>
      <div
        className="profile-experience-block profile-pending-status"
      >
        <div
          className="relative"
        >
          <div
            className="Avatar Avatar-white Avatar-size-109"
          >
            <img
              alt="logo"
              src="appii-placeholder.png"
            />
          </div>
          <p
            className="visible-xs visible-sm"
          >
            <i
              className="profile-status-ico tooltip-position-top tooltip-movable tooltip-warning"
              data-tooltip="Pending verification"
            >
               
            </i>
          </p>
        </div>
        <div
          className="profile-experience-info"
        >
          <i
            className="profile-status-ico tooltip-position-top tooltip-movable tooltip-warning hidden-xs hidden-sm"
            data-tooltip="Pending verification"
          >
             
          </i>
          <p>
            <a
              href="/organisations/953"
              onClick={[Function]}
              title="Oxford Brookes University"
            >
              Oxford Brookes University
            </a>
          </p>
          <span>
            
          </span>
          <em>
            2011
             - 
            2012
          </em>
        </div>
      </div>
    </div>
  </div>
</section>
`;

exports[`containers/Profile/components/EducationList when props.editMode is true when sorting by end_date renders correctly 1`] = `
<section
  className="profile-experience-wrapper"
>
  <div
    className="profile-experience-block-btns"
  >
    <button
      className="reorder-btn"
      onClick={[Function]}
    >
       
    </button>
    <button
      className="add-btn"
      onClick={[Function]}
    >
       
    </button>
  </div>
  <p
    className="profile-block-title"
  >
    Education
  </p>
  <div>
    <div>
      <div
        className="profile-experience-block profile-verified-status"
      >
        <div
          className="profile-experience-btns"
        >
          <button
            className="edit-btn"
            onClick={[Function]}
          >
             
          </button>
        </div>
        <div
          className="relative"
        >
          <div
            className="Avatar Avatar-white Avatar-size-109"
          >
            <img
              alt="logo"
              src="appii-placeholder.png"
            />
          </div>
          <p
            className="visible-xs visible-sm"
          >
            <i
              className="profile-status-ico tooltip-position-top tooltip-movable tooltip-success"
              data-tooltip="Verified"
            >
               
            </i>
          </p>
        </div>
        <div
          className="profile-experience-info"
        >
          <i
            className="profile-status-ico tooltip-position-top tooltip-movable tooltip-success hidden-xs hidden-sm"
            data-tooltip="Verified"
          >
             
          </i>
          <p>
            <a
              href="/organisations/140"
              onClick={[Function]}
              title="Birmingham City University"
            >
              Birmingham City University
            </a>
          </p>
          <span>
            
          </span>
          <em>
            2013
             - 
            2014
          </em>
        </div>
      </div>
    </div>
    <div>
      <div
        className="profile-experience-block profile-pending-status"
      >
        <div
          className="profile-experience-btns"
        >
          <button
            className="edit-btn"
            onClick={[Function]}
          >
             
          </button>
        </div>
        <div
          className="relative"
        >
          <div
            className="Avatar Avatar-white Avatar-size-109"
          >
            <img
              alt="logo"
              src="appii-placeholder.png"
            />
          </div>
          <p
            className="visible-xs visible-sm"
          >
            <i
              className="profile-status-ico tooltip-position-top tooltip-movable tooltip-warning"
              data-tooltip="Pending verification"
            >
               
            </i>
          </p>
        </div>
        <div
          className="profile-experience-info"
        >
          <i
            className="profile-status-ico tooltip-position-top tooltip-movable tooltip-warning hidden-xs hidden-sm"
            data-tooltip="Pending verification"
          >
             
          </i>
          <p>
            <a
              href="/organisations/953"
              onClick={[Function]}
              title="Oxford Brookes University"
            >
              Oxford Brookes University
            </a>
          </p>
          <span>
            
          </span>
          <em>
            2011
             - 
            2012
          </em>
        </div>
      </div>
    </div>
  </div>
</section>
`;
